### PR TITLE
vis + gate algo refactor

### PR DESCRIPTION
**Pull request by dekahaedra**

### CODE REVIEW

File: perception/tasks/gate/GateCenterAlgo.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+from perception.tasks.gate.GateSegmentationAlgoA import GateSegmentationAlgoA
+from perception.tasks.TaskPerceiver import TaskPerceiver
+from typing import Tuple
+from collections import namedtuple
+import sys
+import os
+sys.path.append(os.path.dirname(__file__))
```
Comment (AVSurfer123) [inline perception/tasks/gate/GateCenterAlgo.py]: Remove this. Shouldn't be needed with using absolute imports

File: perception/tasks/gate/GatePerceiver.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import numpy as np
 import sys
 sys.path.insert(0, '..')
```
Comment (AVSurfer123) [inline perception/tasks/gate/GatePerceiver.py]: Again, remove this as shouldn't be needed with absolute imports

File: perception/tasks/gate/GateSegmentationAlgoA.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+from perception.tasks.TaskPerceiver import TaskPerceiver
+from typing import Tuple
+import sys
+import os
+sys.path.append(os.path.dirname(__file__))
```
Comment (AVSurfer123) [inline perception/tasks/gate/GateSegmentationAlgoA.py]: Same thing. We should remove all path modifying lines now.

File: perception/tasks/gate/GateSegmentationAlgoB.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+from typing import Tuple
+import sys
+import os
+sys.path.append(os.path.dirname(__file__))
```
Comment (AVSurfer123) [inline perception/tasks/gate/GateSegmentationAlgoB.py]: Same thing

File: perception/tasks/gate/GateSegmentationAlgoC.py [hunk 5]
```diff
@@ -1,45 +1,45 @@
-from TaskPerceiver import TaskPerceiver
+from perception.tasks.TaskPerceiver import TaskPerceiver
 from typing import Tuple
 import sys
 import os
 from pathlib import Path
 from collections import namedtuple
 sys.path.append(str(Path(__file__).parents[2]) + '/tasks')
```
Comment (AVSurfer123) [inline perception/tasks/gate/GateSegmentationAlgoC.py]: same thing

File: requirements.txt [hunk 6]
```diff
@@ -1,6 +0,0 @@
-opencv-python==3.4.3.18
```
Comment (AVSurfer123) [inline requirements.txt]: keep this and matplotlib only i think

**Review by AVSurfer123 — CHANGES_REQUESTED** [review]

**Comment by AVSurfer123** [comment]
Also why are there 3 different algorithms, A, B, C? is it just older versions? Probably should be renamed better

File: perception/vis/vis.py [hunk 7]
```diff
@@ -1,58 +1,68 @@
...
+def run(data_sources, algorithm, save_video=False):
+    out = None
+    window_builder = Visualizer(algorithm.variables)
+    data = FrameWrapper(data_sources, 0.25)
+    frame_count = 0
+    paused = False
+    speed = 1
+
     for frame in data:
+        if frame_count % speed == 0 and not paused:
```
Comment (nLevin13) [inline perception/vis/vis.py]: If the program is paused, the loop will still run and a lot of frames will be skipped

**Review by nLevin13 — CHANGES_REQUESTED** [review]
I noticed that this branch is several commits behind master, so there's a features I added to vis.py that aren't in your version, namely the pyplot integration. Can you merge from master first then apply your changes? Other than that, it seems good

**Review by nLevin13 — APPROVED** [review]

File: perception/tasks/cross/cross_detection.py [hunk 8]
```diff
@@ -8,10 +8,10 @@
 #############################################################################
 
 sys.path.insert(0, '../background_removal')
```
Comment (AVSurfer123) [inline perception/tasks/cross/cross_detection.py:10]: delete this line

File: perception/tasks/gate/GateSegmentationAlgoB.py (hunk 4 above)
Comment (AVSurfer123) [reply perception/tasks/gate/GateSegmentationAlgoB.py]: I don't see this line deleted. Is this needed for some reason still?

**Review by AVSurfer123 — APPROVED** [review]
Just 2 more small deletions and its good to go

**Comment by nLevin13** [comment]
We can remove (archive) detectGate, threshTest, and threshslider

**Pull request merged**